### DISCUSSION TITLE
unable to select radio buttons

### DISCUSSION THREAD

**Discussion by d-opri** [neuefische/questions#468] [Web] [answered]
moin moin,

i am working on a feature that allows a user to enter names and then select which one paid for the bill. for that i am using a text input to add new objects into the list state. then i am mapping through the array to create the new radio input options. it works,..... except that i cannot select between the radio options. should i not use a boolean value for the checked option? if anyone has a clue i would appreciate the help 

```
function AddParticipants() {
  const [list, setList] = useState([]);

  const [addName, setAddName] = useState("");

  const [isPaid, setIsPaid] = useState(false);

  function handleAddList(e) {
    e.preventDefault();
    setAddName("");
    setList([
      ...list,
      {
        label: addName,
        value: isPaid,
      },
    ]);
  }

  function handleChange() {
    setIsPaid(!isPaid);
  }

  return (
    <>
      <label htmlFor='input'></label>
      <input
        value={addName}
        name='input'
        onChange={(e) => setAddName(e.target.value)}
        type='text'
        placeholder='Add Name'
      />
      <button onClick={handleAddList}>Add Participant</button>

      <Subtitle>Paid By</Subtitle>
      <br />

      {list.map((item) => (
        <>
          <Layout>
            <label htmlFor='participants'>{item.label}</label>
            <input
              minLength={2}
              type='radio'
              value={item.label}
              onChange={handleChange}
              name='participants'
              checked={item.value}
            />
          </Layout>
        </>
      ))}
    </>
  );
}
```

**Comment by doemser** [comment, marked as answer]
Here is an isolated prototype of the approach you are looking for:

https://codesandbox.io/s/romantic-dew-xvivkm?file=/src/App.js

In this approach every radio button uses its value from the selectedPerson state, so you can only ever select one. Hope this helps you already.

**Reply by d-opri** [reply]
thanks a lot! that fixed my issue :)
